contribs/src/main/scala/io/delta/storage/BaseExternalLogStore.scala

```diff
@@ -0,0 +1,309 @@
+        case e: Throwable =>
+          logInfo(s"${e.getClass.getName}: $e")
+          if (retry >= 3) throw e
+      }
+      retry += 1;
+    }
+  }
+
+  /**
+   * Returns path stripped user info.
+   */
+  protected def getPathKey(resolvedPath: Path): Path = {
+    stripUserInfo(resolvedPath)
+  }
+
+  /**
+   * Generate temporary path for TransactionLog.
+   */
+  protected def createTemporaryPath(path: Path): String = {
+    val uuid = java.util.UUID.randomUUID().toString
+    s".tmp/${path.getName}.$uuid" // Q: remove path.getName part?
+  }
+  override def listFrom(path: Path): Iterator[FileStatus] = {
+    listFrom(path, getHadoopConfiguration)
+  }
+
+  override def listFrom(
+      path: Path,
+      hadoopConf: Configuration
+  ): Iterator[FileStatus] = {
+    logDebug(s"listFrom path: ${path}")
+    val (fs, resolvedPath) = resolvePath(path, hadoopConf)
+    var entry = getLatestDbEntry(resolvedPath.getParent)
+    entry.foreach(fixDeltaLog(fs, _))
+
+    val parentPath = resolvedPath.getParent
+    if (!fs.exists(parentPath)) {
+      throw new FileNotFoundException(s"No such file or directory: $parentPath")
+    }
+
+    super.listFrom(path, hadoopConf)
+  }
+
+  /**
+   * Write file with actions under a specific path.
+   */
+  protected def writeActions(
+      fs: FileSystem,
+      path: Path,
+      actions: Iterator[String]
+  ): Unit = {
+    logDebug(s"writeActions to: $path")
+    val stream = fs.create(path, true)
+    actions.map(_ + "\n").map(_.getBytes(UTF_8)).foreach(stream.write)
+    stream.close()
+  }
+
+  def write(
+      path: Path,
+      actions: Iterator[String],
+      overwrite: Boolean = false
+  ): Unit = {
+    write(path, actions, overwrite, getHadoopConfiguration)
+  }
+
+  override def write(
+      path: Path,
+      actions: Iterator[String],
+      overwrite: Boolean,
+      hadoopConf: Configuration
+  ): Unit = {
+    val (fs, resolvedPath) = resolvePath(path, hadoopConf)
+
+    if (overwrite) {
+      return writeActions(fs, path, actions);
+    };
+
+    val parentPath = resolvedPath.getParent
+    assert(FileNames.isDeltaFile(path))
+    val version = FileNames.deltaVersion(path)
+
+    if (version > 0) {
+      val prevVersion = version - 1
+      val prevPath = FileNames.deltaFile(parentPath, prevVersion)
+      getDbEntry(prevPath) match {
+        case Some(entry) => fixDeltaLog(fs, entry)
+        case None =>
+          if (!fs.exists(prevPath)) {
+            throw new java.nio.file.FileSystemException(
+              s"previous commit ${prevPath} doesn't exist"
+            )
+          }
+          // previous commit exists in fs but not in dynamodb
+      }
+    } else {
+      getDbEntry(path) match {
+        case Some(entry) =>
+          if (entry.complete && !fs.exists(path)) {
+            throw new java.nio.file.FileSystemException(
+              s"Old entries for ${parentPath} still exist in the database"
+            )
+          }
+        case None => ;
+      }
+    }
```

Comment:
all of this can be wrapped in a `FileNames.isDeltaFile(path)` if condition.
if this is not a delta file, and `overwrite=false`, then we do not need to (and cannot) check for the `prevVersion`. this code will then make less assumptions about the method input.

Reply:
makes sense

Reply:
It is really great idea! Just tested it and it seems I simply can revert my all changes to LogStoreSuite.scala
Thanks!